### PR TITLE
Generic generated message

### PR DESCRIPTION
**Pull request by sigurdm**
This would solve https://github.com/dart-lang/protobuf/issues/177 .
We have to measure the impact on existing codebases before landing this.

### CODE REVIEW

File: protoc_plugin/pubspec.yaml [hunk 1]
```diff
@@ -18,3 +18,7 @@ dev_dependencies:
 
 executables:
   protoc-gen-dart: protoc_plugin
+
+dependency_overrides:
```
Comment (szakarias) [inline protoc_plugin/pubspec.yaml]: remove :-)

File: protobuf/lib/src/protobuf/builder_info.dart [hunk 2]
```diff
@@ -17,9 +17,11 @@ class BuilderInfo {
   bool hasExtensions = false;
   bool hasRequiredFields = true;
   List<FieldInfo> _sortedByTag;
+  CreateBuilderFunc<M> _builder;
 
-  BuilderInfo(String messageName, {PackageName package = const PackageName('')})
-      : qualifiedMessageName = "${package.prefix}$messageName";
+  BuilderInfo(String messageName, {CreateBuilderFunc<M> builder, PackageName package = const PackageName('')})
```
[the remark this answers is not in the document]
Comment (sigurdm) [reply protobuf/lib/src/protobuf/builder_info.dart]: Agree. Fixed

File: protobuf/lib/src/protobuf/pb_map.dart [hunk 3]
```diff
@@ -4,6 +4,21 @@
...
+class _MapEntry<K, V> extends GeneratedMessage<_MapEntry<K, V>> {
+  _MapEntry(int keyFieldType, int valueFieldType, CreateBuilderFunc valueCreator, ValueOfFunc valueOf, List<ProtobufEnum> enumValues);
+
+  @override
+  _MapEntry<K, V> clone() {
+    throw "Should not be instantiated";
+  }
+
+  @override
+  // TODO: implement info_
```
[the remark this answers is not in the document]
Comment (sigurdm) [reply protobuf/lib/src/protobuf/pb_map.dart]: Done

File: protobuf/test/builder_info_test.dart [hunk 4]
```diff
@@ -5,17 +5,23 @@
 import 'package:protobuf/protobuf.dart';
 import 'package:test/test.dart';
 
+class Foo extends GeneratedMessage<Foo> {
+  @override
+  // TODO: implement info_
```
[the remark this answers is not in the document]
Comment (sigurdm) [reply protobuf/test/builder_info_test.dart]: Done

File: protoc_plugin/pubspec.yaml (hunk 1 above)
Comment (sigurdm) [reply protoc_plugin/pubspec.yaml]: Done

**Comment by listepo** [comment]
Hello @sigurdm, any news for this PR?

**Comment by sigurdm** [comment]
I want to do some measurements of the performance impact before landing this.
Unfortunately it might still be a few weeks in the future before I get around to it.

**Comment by sigurdm** [comment]
I will not get time to land this in the short future.
Closing the PR for now.

**Pull request closed without merging**